tensorrt_llm/_torch/pyexecutor/mamba_cache_manager.py

```
@@ -1300,9 +1339,16 @@ def update_resources(self,
-            self.impl.copy_linear_attention_block(req)
-        self.impl.refresh_blocks()
+        # Issue all async block onboards; defer the syncTransfers() (refresh_blocks)
+        # until just before forward so the CPU-side prep (attn metadata, input
+        # tensor assembly, draft model prep, etc.) can overlap with the in-flight
+        # cudaMemcpyAsync calls instead of being serialized behind them.
+        # copy_linear_attention_block returns True iff it actually issued a copy;
+        # we skip refresh_blocks entirely when nothing was scheduled.
+        self._pending_state_transfers = self.impl.copy_linear_attention_block_batch(
+            self.requests)
+        if self._pending_state_transfers:
+            logger.info(f"Need to transfer mamba state blocks")
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)